src/App.tsx

```diff
@@ -1,14 +1,63 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo, FilterType } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterType, setFilterType] = useState(FilterType.ALL);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(todo => {
+        setTodos(todo);
+        setIsLoading(true);
+      });
+  }, [selectedTodo?.userId]);
+
+  const filterTodo = useMemo(() => {
+    return todos.filter(todo => {
+      const todosFilter = todo.title.toLowerCase().includes(query.toLowerCase());
+
+      switch (filterType) {
+        case FilterType.ALL:
+          return todosFilter;
+
+        case FilterType.COMPLETE:
+          return todo.completed && todosFilter;
+
+        case FilterType.ACTIVE:
+          return !todo.completed && todosFilter;
+
+        default:
+          return todosFilter;
+      }
+    });
+  }, [query, todos, filterType]);
+
+  const onChangeQuery = useCallback((e) => {
```

Comment:
useCallback doesn't memorize when child component doesn't wrap in memo, delete useCallback's or wrap  TodoFilter into memo